### ISSUE TITLE
Use Jackson annotations for enum values

### ISSUE THREAD

**Issue by ronan-skyscanner** [victools/jsonschema-generator#27]
From the discussion in https://github.com/victools/jsonschema-generator/issues/25

There can be cases where an enum uses a toString() method representation, and a different format for the serialisation to json.
When using Jackson, the latter method is annotated with @JsonValue to instruct Jackson to us that implementation for JSON serialisation.

Currently the library allows using the `name` value, or the `toString()` value (with the new `FLATTENED_ENUMS_FROM_TOSTRING` option), but the Jackson module doesn't correctly use the annotated method to serialise in the schema. This means that the generated schema has different enum values to the json generated.

Is there a way to make use of the JsonValue annotation when generating the schema to keep both in sync?

**Comment by CarstenWickner** [comment]
Hi @ronan-skyscanner,

I don't see any reason why that shouldn't be possible.
In the jackson module, the `@JsonValue` annotation should probably be considered for enums when `FLATTENED_ENUMS` or `FLATTENED_ENUMS_FROM_TOSTRING` is enabled on the main generator. But would also make sense to consider that annotation for any other constant values (if they are included in the generated schema).

Will have to figure out a nice way to handle it and most importantly find the time for that. I currently can't promise when I would get to it, but you're also welcome to create a Pull Request.

**Comment by CarstenWickner** [comment]
HI @ronan-skyscanner,

I've just released version [v4.5.0](https://github.com/victools/jsonschema-module-jackson/releases/tag/v4.5.0) of the `jsonschema-module-jackson`, which contains a single new feature (as per the linked PR above). For convenience sake, I'm copying my description here again:

> Introducing a new `JacksonOption.FLATTENED_ENUMS_FROM_JSONVALUE` that can be provided in the `JacksonModule`'s constructor.
> When included, any enum with a single eligible `@JsonValue` annotated method will be treated as a string as per standard `Option.FLATTENED_ENUMS`, but invoking the annotated method instead of `name()` (or `toString()` in case of `Option.FLATTENED_ENUMS_FROM_TOSTRING`).
> 
> When encountering an enum without such an annotated method, the default handling applies – i.e. it is recommended to combine this new `JacksonOption` with your desired default behavior (either falling-back on `name()` or `toString()`) steered via one of the aforementioned standard `Option` values on the main `jsonschema-generator`.

No change in the main `jsonschema-generator` is necessary.
You can enable the new option by simply instantiating the `JacksonModule` in the following way:
```java
new JacksonModule(JacksonOption.FLATTENED_ENUMS_FROM_JSONVALUE)
```
